### ISSUE TITLE
do not warn if using pattern matching in |a, e|, e.g |a, (digest, ids)|

### ISSUE THREAD

**Issue by 404pnf** [rubocop/rubocop#463]
I take the advice of rubocop very seriously.   But I tend to disagree on situation like this: 

``` Ruby
    find_duplicate_records.rb:38:3: C: Name reduce arguments |a, e| (accumulator, element).
      db.reduce([]) { |a, (_, ids)|  a << ids if ids.size > 1 ; a }
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    find_duplicate_records.rb:38:59: C: Do not use semicolons to terminate expressions.
      db.reduce([]) { |a, (_, ids)|  a << ids if ids.size > 1 ; a }
```

I think by using pattern matching, |a (_, ids)| is better than using |a, e| then call e[1] in the block.
